File: Contest 4/main.cpp
#include <bits/stdc++.h>

using namespace std;

bool isPrime(int n)
{
    for (int i = 0; i < floor(sqrt(n)); i++)
    {
        if (n % i == 0)
        {
            return false;
        }
    }
    return true;
}

bool isPSquare(int n)
{

}

int main()
{   
    int pieceLoc [6];
    int r;

    for (int i = 0; i < 6; i++)
    {
        cin >> pieceLoc[i];
    }
    sort(pieceLoc, pieceLoc + 6);
    cin >> r;

    for (int i = 0; i < r; i++)
    {
        int n;
        cin >> n;

        int temp0 = pieceLoc[0] + n;
        if (temp0 == pieceLoc[1])
        {
            continue;
        }
        else if (temp0 == 52)
        {
            pieceLoc[0] = 999;
        }
        else if (isPrime(temp0))
        {
            for (int i = 0; i < 6; i++)
            {
                if (temp0 < pieceLoc[1]-1)
                {
                    temp0++;
                }
            }
        }

        
        // if (pieceLoc[0] + n != pieceLoc[1])
        // {
        //     pieceLoc[0] += n
        // }
        // for (int j = 0; j < 6; j++)
        // {
        //     cout << pieceLoc[j] << " ";
        // }
        // cout << " \n";
    }

    
}

File: Contest 4/main.py
import math

def main():
    inpt = [int(x) for x in input().split()]
    pieces = inpt[0:6]
    rolls = inpt[7:]

    for r in rolls:
        pieces.sort()
        temp = pieces[0] + r

        if (temp in pieces):
            continue
        elif (isPrime(temp)):
            count = 0
            while (temp < pieces[inbetween(temp, pieces)[1]] - 1 and count < 6):
                temp += 1
        elif (temp in [9, 16, 25, 36, 49]):
            while (temp > pieces[inbetween(temp, pieces)[0]] + 1 and count < 6):
                temp -= 1
        elif (passedCorner(pieces[0], temp)):
            while (temp > pieces[0] and temp % r != 0 or temp in pieces):
                temp -= 1
        
        if (temp == 52):
            pieces[0] = 999
        elif (temp > 52):
            continue
        else:
            pieces[0] = temp
    pieces.sort()
    print(pieces)
        


def inbetween(n, l):
    for count, i in enumerate(l):
        if(i > n):
            return ([count-1, count])
    return ([-1, 1])
        
def isPrime(n):
    for i in range(2, math.floor(math.sqrt(n)) + 1):
        if (n % i == 0):
            return False
    return True

def passedCorner(n1, n2):
    corners = [7, 12, 17, 22, 35, 40, 45, 50]
    if(inbetween(n1, corners) != inbetween(n2, corners)):
        return True
    return False

main()